drivers/ieee802154/ieee802154_efr32.c

```diff
@@ -0,0 +1,572 @@
+/* ieee802154_efr32.c - Silabs EFR32 driver */
+
+/*
+ * Copyright (c) 2018 Evry ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME ieee802154_efr32
+#define LOG_LEVEL CONFIG_IEEE802154_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <irq.h>
+#include <net/ieee802154_radio.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+#include <misc/byteorder.h>
+#include <random/rand32.h>
+
+#include "rail_types.h"
+#include "em_core.h"
+#include "em_system.h"
+
+#include "rail.h"
+#include "rail_ieee802154.h"
+#include "pa_conversions_efr32.h"
+
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+#include <net/openthread.h>
+#endif
+
+#define RADIO_CONFIG_XTAL_FREQUENCY 38400000UL
+
+RAIL_DECLARE_TX_POWER_VBAT_CURVES(piecewiseSegments, curvesSg, curves24Hp, curves24Lp);
+
+#define ACK_TIMEOUT K_MSEC(10)
+
+enum EfrState
+{
```

Comment:
{ should be on the same line as enum (same as struct etc...) it's only on actual function implementation that a line break is required.